File: frontend/src/modules/objects/voteObject.js
export default class VoteObject {
  constructor(userid, suggestionid) {
    this.userid = userid;
    this.suggestionid = suggestionid;
  }

  getUserId() {
    return this.userid;
  }

  getSuggestionId() {
    return this.suggestionid;
  }

  setUserId(userid) {
    this.userid = userid;
  }

  setSuggestionId(suggestionid) {
    this.suggestionid = suggestionid;
  }

  static fromJSON(json) {
    let result = [];

    // json.forEach((f) => {
    //   const data = Object.values(f);
    //   console.log(data);
    //   result.push(data);
    // });
    // console.log(result);
    // return result;
    if (Array.isArray(json)) {
      /* Multiple entries */
      json.forEach((f) => {
        Object.setPrototypeOf(f, VoteObject.prototype);
        result.push(f);
      });
    } else {
      /* Single entry */
      let f = json;
      Object.setPrototypeOf(f, VoteObject.prototype);
      result.push(f);
    }

    return result;
  }
}

File: backend/objects/voteObject.py
class VoteObject:
    def __init__(self, userid, suggestionid):
        self._userid = userid
        self._suggestionid = suggestionid

    def get_userid(self):
        return self._userid

    def get_suggestionid(self):
        return self._suggestionid

    def set_userid(self, userid):
        self._userid = userid

    def set_suggestionid(self, suggestionid):
        self._suggestionid = suggestionid

    @staticmethod
    def from_dict(dictionary=dict(), set_id=False):
    
        vote = VoteObject(dictionary["userid"], dictionary["suggestionid"])
        return vote